### PR TITLE
Add support for MRS1000 scanner

### PR DESCRIPTION
**Pull request by ipa-fez**
Adds support for the MRS1000 scanner with 4 scan layers.

Each layer has up to 3 echoes. The node publishes  a PointCloud2 with all four layers and RSSI values for the points as well as LaserScan and MultiEchoLaserScan messages for each layer.

The used echo and the max range can be configured via launch params.

### CODE REVIEW

File: src/colaa_conversion.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+  double start_angle = data.ch16bit[channel].header.start_angle * M_PI / 180.0 / 10000.0 - M_PI / 2.0;
+  double angle_increment = data.ch16bit[channel].header.step_size * M_PI / 180.0 / 10000.0;
+
+  scan.angle_increment = angle_increment;
+  scan.angle_min = start_angle;
+  scan.angle_max = start_angle + (data.ch16bit[0].data.size() - 1) * angle_increment;
+  for (size_t k = 0; k < data.ch16bit[channel].data.size(); ++k)
+  {
+    scan.ranges[k] = data.ch16bit[channel].data[k] * 0.001 * data.ch16bit[channel].header.scale_factor;
```
Comment (jabbenseth) [inline src/colaa_conversion.cpp:37]: how about checking for `range > max_range` here and setting to `inf` if so?

Comment (ipa-fez) [reply src/colaa_conversion.cpp:37]: I think when I implemented this for LMS5xx @ipa-bnm told me this should be handled by the clients.

Comment (jabbenseth) [reply src/colaa_conversion.cpp:37]: ok. So to "handle by the client": https://github.com/ros-perception/laser_filters/pull/56

File: launch/MRS1000.launch [hunk 2]
```diff
@@ -1,8 +1,17 @@
 <launch>
   <arg name="host" default="192.168.2.140" />
   <arg name="port" default="2111" />
+  <arg name="range" default="64.0" />
+  <!-- first, last, all -->
```
Comment (jabbenseth) [inline launch/MRS1000.launch:5]: no "strongest" option here?

Comment (ipa-fez) [reply launch/MRS1000.launch:5]: This is for configuring the sensor, those are the only options the hardware provides. "Strongest" is done in software by the ros node by sorting "all" echoes by reflectance.

**Review by jabbenseth — APPROVED** [review]

**Review by ipa-fke — APPROVED** [review]
Leave a comment

**Pull request merged**